feat(LocalePack): read the content of the given locale pack file

ksza opened the pull request:
At this point the client apps using this library are injecting a `<script type='text/javascript' src='' />` tag in the webpage which then downloads the locale pack. This has an impact on the initial page load.

With this new helper, clients can choose to read the content of the locale file and serve the localePack preloaded on the window.

Commit:
feat(LocalePack): read the content of the given locale pack file

corthmann commented:
Thank you @ksza :heart: 

Yes there is a trade-off here between initial page load performance and subsequent page load performance. Using the existing code all secondary page-views require less data transferred (assuming the compiled packs are given proper caching headers).

I think the feature you request makes sense and feel we should add support for it in `locale_pack`.

Looking at the implementation in the PR, then I see two main problems:

1) reading the file contents from the file system on every request is also expensive. I think we should be able to find a compromise where contents of files used by this helper will be memoized.
2) the code works only for people using the default configuration (and rails) due to the hard-coded "public" part of the read path. The helper will not work for projects that have configured their `output_path`.

I'll give some thought to the problem and get back to you as soon as possible.

ksza commented:
> Thank you @ksza ❤️
> 
> Yes there is a trade-off here between initial page load performance and subsequent page load performance. Using the existing code all secondary page-views require less data transferred (assuming the compiled packs are given proper caching headers).
> 
> I think the feature you request makes sense and feel we should add support for it in `locale_pack`.
> 
> Looking at the implementation in the PR, then I see two main problems:
> 
> 1. reading the file contents from the file system on every request is also expensive. I think we should be able to find a compromise where contents of files used by this helper will be memoized.
> 2. the code works only for people using the default configuration (and rails) due to the hard-coded "public" part of the read path. The helper will not work for projects that have configured their `output_path`.
> 
> I'll give some thought to the problem and get back to you as soon as possible.

Thanks @corthmann. Optimising for subsequent reads is certainly necessary and required to fit production purposes. 

Thanks for taking time to look into this and looking forward for your suggestions

Commit:
feat(LocalePack): memoize file content and use configured output path

Commit:
feat(LocalePack): remove comments

ksza commented:
@corthmann please check this pull req again. I've addressed your comments by caching subsequent request to the same file and by using the configured output_path

ksza commented:
closing this as there seems to be a lack of interest in the proposal
